Source/SwiftLintBuiltInRules/Rules/Lint/PrivateSwiftUIStatePropertyRule.swift

```diff
@@ -0,0 +1,246 @@
+            class Foo: ObservableObject {
+                @State var bar = Bar()
+            }
+            """),
+            Example("""
+            extension MyObject {
+                struct ContentView: View {
+                    @State private var isPlaying: Bool = false
+                }
+            }
+            """),
+            Example("""
+            actor ContentView: View {
+                @State private var isPlaying: Bool = false
+            }
+            """)
+        ],
+        triggeringExamples: [
+            Example("""
+            struct MyApp: App {
+                @State ↓var isPlaying: Bool = false
+            }
+            """),
+            Example("""
+            struct MyScene: Scene {
+                @State ↓var isPlaying: Bool = false
+            }
+            """),
+            Example("""
+            struct ContentView: View {
+                @State ↓var isPlaying: Bool = false
+            }
+            """),
+            Example("""
+            struct ContentView: View {
+                struct InnerView: View {
+                    @State private var showsIndicator: Bool = false
+                }
+
+                @State ↓var isPlaying: Bool = false
+            }
+            """),
+            Example("""
+            struct MyStruct {
+                struct ContentView: View {
+                    @State ↓var isPlaying: Bool = false
+                }
+            }
+            """),
+            Example("""
+            struct MyStruct {
+                struct ContentView: View {
+                    @State ↓var isPlaying: Bool = false
+                }
+
+                @State var isPlaying: Bool = false
+            }
+            """),
+            Example("""
+            final class ContentView: View {
+                @State ↓var isPlaying: Bool = false
+            }
+            """),
+            Example("""
+            extension MyObject {
+                struct ContentView: View {
+                    @State ↓var isPlaying: Bool = false
+                }
+            }
+            """),
+            Example("""
+            actor ContentView: View {
+                @State ↓var isPlaying: Bool = false
+            }
+            """)
+        ]
+    )
+
+    func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
+
+private extension PrivateSwiftUIStatePropertyRule {
+    final class Visitor: ViolationsSyntaxVisitor {
+        override var skippableDeclarations: [DeclSyntaxProtocol.Type] {
+            [ProtocolDeclSyntax.self]
+        }
+
+        /// LIFO stack that is stores type inheritance clauses for each visited node
+        /// The last value is the inheritance clause for the most recently visited node
+        /// A nil value indicates that the node does not provide any inheritance clause
+        private var visitedTypeInheritances: [TypeInheritanceClauseSyntax?] = []
```

Comment:
There is a `Stack` type in SwiftLint specifically for such use cases.

Reply:
Nice! Updated